src/state/download/actions.js

```diff
@@ -246,26 +246,68 @@ export const fetchDataSetSucceeded = (
   }
 });
 
-export const fetchDataSetDetails = dataSet => {
-  return async dispatch => {
-    dispatch({
-      type: 'DOWNLOAD_FETCH_DETAILS'
-    });
-    const { experiments, samples } = await getSamplesAndExperiments(dataSet);
-    dispatch(fetchDataSetDetailsSucceeded(experiments, samples));
-  };
-};
+export const fetchDataSetDetails = () => async dispatch => {
+  const dataSetId = localStorage.getItem('dataSetId');
+  if (!dataSetId) {
+    return;
+  }
 
-export const fetchDataSetDetailsSucceeded = (experiments, samples) => {
-  return {
-    type: 'DOWNLOAD_FETCH_DETAILS_SUCCESS',
+  dispatch({
+    type: 'DOWNLOAD_FETCH_DETAILS',
     data: {
-      experiments,
-      samples
+      dataSetId
     }
-  };
+  });
+  let {
+    data,
+    is_processing,
+    is_processed,
+    aggregate_by,
+    scale_by,
+    samples,
+    experiments
+  } = await getDataSetDetails(dataSetId);
+
+  [samples, experiments] = formatSamplesAndExperiments(
+    data,
+    samples,
+    experiments
+  );
+
+  dispatch(
+    fetchDataSetDetailsSucceeded(
+      data,
+      is_processing,
+      is_processed,
+      aggregate_by,
+      scale_by,
+      samples,
+      experiments
+    )
+  );
 };
 
+export const fetchDataSetDetailsSucceeded = (
+  dataSet,
+  is_processing,
+  is_processed,
+  aggregate_by,
+  scale_by,
+  samples,
+  experiments
```

Comment:
It might be better to use [an object](https://github.com/AlexsLemonade/refinebio-frontend/pull/51#discussion_r192516899) as a parameter

Reply:
I agree, since the input arguments are exactly same as the value of `data` in the returned object.

Reply:
Ups, I missed this. Will upload the change in a sec, sorry!

Reply:
When you said `use an object as a parameter`, I thought you meant this:
```
export const fetchDataSetDetailsSucceeded = dataObj => { type: 'DOWNLOAD_FETCH_DETAILS_SUCCESS', ...dataObj }
```
Am I too crazy?

Reply:
hahah nah I don't think so... This is an interesting approach that I hadn't considered, although in my opinion, in this case listing the variables that we're passing helps with the readability of the code. We write these variables in the reducer as well, just to emphasize what we're passing https://github.com/AlexsLemonade/refinebio-frontend/blob/5d4e6c5c21a64d77324a2337b377f2ba644f1d30/src/state/download/reducer.js#L70-L79

In any case, both would be equivalent, and yours is definitely shorter! I can change it if you want.